community/kernel/src/test/java/org/neo4j/kernel/impl/transaction/log/TransactionLogAppendAndRotateStressIT.java

```diff
@@ -0,0 +1,276 @@
+        TransactionMetadataCache metadataCache = new TransactionMetadataCache( 10, 100 );
+        LogFile logFile = life.add( new PhysicalLogFile( fs, logFiles, rotationThreshold,
+                txIdStore, logVersionRepository, monitoring, metadataCache ) );
+        monitoring.setLogFile( logFile );
+        KernelHealth health = new KernelHealth( mock( KernelPanicEventGenerator.class ), NullLog.getInstance() );
+        LogRotation rotation = new LogRotationImpl( monitoring, logFile, health );
+        final TransactionAppender appender = life.add( new BatchingTransactionAppender( logFile, rotation, metadataCache,
+                txIdStore, BYPASS, health ) );
+
+        // WHEN
+        Race race = new Race();
+        for ( int i = 0; i < 10; i++ )
+        {
+            race.addContestant( new Runnable()
+            {
+                @Override
+                public void run()
+                {
+                    while ( !end.get() )
+                    {
+                        try
+                        {
+                            appender.append( sillyTransaction( 1_000 ), NULL );
+                        }
+                        catch ( Exception e )
+                        {
+                            e.printStackTrace( System.out );
+                            end.set( true );
+                            fail( e.getMessage() );
+                        }
+                    }
+                }
+            } );
+        }
+        race.addContestant( endAfterMax( 10, SECONDS, end ) );
+        race.go();
+
+        // THEN
+        assertTrue( monitoring.numberOfRotations() > 0 );
+    }
+
+    private Runnable endAfterMax( final int time, final TimeUnit unit, final AtomicBoolean end )
+    {
+        return new Runnable()
+        {
+
+            @Override
+            public void run()
+            {
+                long endTime = currentTimeMillis() + unit.toMillis( time );
+                while ( currentTimeMillis() < endTime && !end.get() )
+                {
+                    parkNanos( MILLISECONDS.toNanos( 50 ) );
+                }
+                end.set( true );
+            }
+        };
+    }
+
+    private static void assertWholeTransactionsIn( LogFile logFile, long logVersion ) throws IOException
+    {
+        try ( ReadableLogChannel reader = logFile.getReader( new LogPosition( logVersion, LOG_HEADER_SIZE ) ) )
+        {
+            VersionAwareLogEntryReader<ReadableLogChannel> entryReader = new VersionAwareLogEntryReader<>();
+            LogEntry entry = null;
+            int state = 0;
+            int transactions = 0;
+            while ( (entry = entryReader.readLogEntry( reader )) != null )
+            {
+                switch ( state )
+                {
+                case 0: // Expects start entry
+                    assertTrue( entry instanceof LogEntryStart );
+                    state = 1;
```

Comment:
that's true but with counter we will check that number of entry start is the same as number of commits